File: frontend/lib/models/task.dart
class Task {
  String title;
  bool completed;
  String category;

  Task({
    required this.title,
    this.completed = false,
    this.category = "Personal",
  });

  Map<String, dynamic> toJson() {
    return {
      "title": title,
      "completed": completed,
      "category": category,
    };
  }

  factory Task.fromJson(
      Map<String, dynamic> json) {
    return Task(
      title: json["title"],
      completed: json["completed"] ?? false,
      category: json["category"] ?? "Personal",
    );
  }
}

File: task_master_backend/models/Task.js
const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },

    title: {
      type: String,
      required: true,
    },

    completed: {
      type: Boolean,
      default: false,
    },

    category: {
      type: String,
      default: "Personal",
    },

    dueDate: {
      type: String,
      default: "No Date",
    },
  },
  {
    timestamps: true,
  }
);

module.exports =
  mongoose.models.Task ||
  mongoose.model("Task", taskSchema);